src/ImportCsvDialog.cpp

```diff
@@ -412,6 +449,55 @@ void ImportCsvDialog::importCsv(const QString& fileName)
     }
 }
 
+sqlb::FieldVector ImportCsvDialog::fetchCsvHeader(const QString& fileName)
```

Comment:
This function is more or less a copy of ```importCsv()```. I think we should modify it slightly, so it can be called from inside ```importCsv()``` as well to share the common code. Maybe change it to take a pointer to a CSVParser object and use that instead of taking a file name and opening the file etc. Though I'm not exactly sure how to do that ideally...